### ISSUE TITLE
hardware.nvidiaOptimus.disable no longer works

### ISSUE THREAD

**Issue by peti** [NixOS/nixpkgs#28333]
Current versions of `nixos-unstable` (like 129f8d7e99) can no longer load the `bbswitch` module on my Latitude E7450 laptop:

~~~
$ modprobe -v bbswitch                                                  root/git/master 
insmod /run/current-system/kernel-modules/lib/modules/4.9.43/misc/bbswitch.ko 
modprobe: ERROR: could not insert 'bbswitch': No such device
~~~

The systemd journal says:

~~~~
kernel: bbswitch: version 0.8
kernel: bbswitch: Found integrated VGA device 0000:00:02.0: \_SB_.PCI0.GFX0
kernel: bbswitch: Found discrete VGA device 0000:03:00.0: \_SB_.PCI0.RP05.D034
kernel: bbswitch: failed to evaluate \_SB_.PCI0.RP05.D034._DSM {0xF8,0xD8,0x86,0xA4,0xDA,0x0B,0x1B,0x47,0xA7,0x2B,0x60,0x42,0xA6,0xB5,0xBE,0xE0} 0x100 0x0 {0x00,0x00,0x00,0x00}: AE_NOT_FO
kernel: bbswitch: failed to evaluate \_SB_.PCI0.RP05.D034._DSM {0xA0,0xA0,0x95,0x9D,0x60,0x00,0x48,0x4D,0xB3,0x4D,0x7E,0x5F,0xEA,0x12,0x9F,0xD4} 0x102 0x0 {0x00,0x00,0x00,0x00}: AE_NOT_FO
kernel: bbswitch: failed to evaluate \_SB_.PCI0.GFX0._DSM {0xA0,0xA0,0x95,0x9D,0x60,0x00,0x48,0x4D,0xB3,0x4D,0x7E,0x5F,0xEA,0x12,0x9F,0xD4} 0x102 0x0 {0x00,0x00,0x00,0x00}: AE_NOT_FOUND
kernel: bbswitch: No suitable _DSM call found.
~~~~

This used to work fine in previous versions of NixOS.

**Comment by peti** [comment]
`bbswitch` works fine in a7c8f5e419.

**Comment by peti** [comment]
Switching to the latest Linux Kernel (4.12.7) works around the issue.

**Comment by tilpner** [comment]
This doesn't seem fixed in 17.09, on Linux 4.12.10. The modprobe works, but when trying to boot with `hardware.nvidiaOptimus.disable = true;` my laptop freezes and requires a forced reboot.

The logs show nothing mentioning either bbswitch or nvidia, not even something like `bbswitch: version 0.8` from the OP

**Comment by peti** [comment]
I'm closing this issue since I no longer have it so it's fixed, I guess.
